### PR TITLE
Feature/pt 1005 openapi doc

### PR DESCRIPTION
**Pull request by BigFab**
I was not able to have the nested types generated as a definition in the output documentation. For example, the returned _status.ApiStatus_ is defined as {version: string, status: Status} where _Status_ does not have his own definition (which is supposed to be a {Code, Reason}) in the generated documentation. This looks like a drawback of swaggo for which I did not find any solution.

### CODE REVIEW

File: data/api.go [hunk 1]
```diff
@@ -127,24 +126,26 @@ func (a *API) GetStatus(res http.ResponseWriter, req *http.Request) {
...
-// latest (optional) : Returns only the most recent results for each `type` matching the results filtered by the other query parameters
 // @Summary Get device/health data for a user based on a set of parameters
 // @Description Get device/health data for a user based on a set of parameters
 // @ID tide-whisperer-api-getdata
+// @Produce json
+// @Success 200
+// @Failure 500 {object} data.detailedError
+// @Failure 403 {object} data.detailedError
+// @Param userID path int true "The ID of the user to search data for"
+// @Param type query string false "Type of data to search for - can be a list of types separated by commas"
```
Comment (CedricPerotto) [inline data/api.go:137]:
```suggestion
// @Param type query []string false "Type of data to search for - can be a list of types separated by commas"
```

File: data/api.go [hunk 2]
```diff
@@ -127,24 +126,26 @@ func (a *API) GetStatus(res http.ResponseWriter, req *http.Request) {
...
+// @Param subType query string false "Subtype of data to search for - can be a list of subtypes separated by commas"
```
Comment (CedricPerotto) [inline data/api.go:138]:
```suggestion
// @Param subType query []string false "Subtype of data to search for - can be a list of subtypes separated by commas"
```

File: data/api.go [hunk 3]
```diff
@@ -127,24 +126,26 @@ func (a *API) GetStatus(res http.ResponseWriter, req *http.Request) {
...
+// @Param deviceId query string false "ID of the device to search data for"
+// @Param uploadId query string false "ID of the upload to search data for"
+// @Param startDate query string false "ISO Date time for search lower limit"
```
Comment (CedricPerotto) [inline data/api.go:141]:
```suggestion
// @Param startDate query string false "ISO Date time for search lower limit" format(date-time)
```

File: data/api.go [hunk 4]
```diff
@@ -127,24 +126,26 @@ func (a *API) GetStatus(res http.ResponseWriter, req *http.Request) {
...
+// @Param endDate query string false "ISO Date time for search upper limit"
```
Comment (CedricPerotto) [inline data/api.go:142]:
```suggestion
// @Param endDate query string false "ISO Date time for search upper limit" format(date-time)
```

File: data/api.go [hunk 5]
```diff
@@ -127,24 +126,26 @@ func (a *API) GetStatus(res http.ResponseWriter, req *http.Request) {
...
-// startDate (optional) : Only objects with 'time' field equal to or greater than start date will be returned.
-//					Must be in ISO date/time format e.g. 2015-10-10T15:00:00.000Z
-// endDate (optional) : Only objects with 'time' field less than to or equal to start date will be returned.
-//					Must be in ISO date/time format e.g. 2015-10-10T15:00:00.000Z
-// latest (optional) : Returns only the most recent results for each `type` matching the results filtered by the other query parameters
 // @Summary Get device/health data for a user based on a set of parameters
 // @Description Get device/health data for a user based on a set of parameters
 // @ID tide-whisperer-api-getdata
+// @Produce json
+// @Success 200
```
Comment (CedricPerotto) [inline data/api.go:133]:
```suggestion
// @Success 200 {array} deviceData "List of user data objects"
```

File: data/api.go [hunk 6]
```diff
@@ -102,10 +102,9 @@ func (a *API) Get501(res http.ResponseWriter, req *http.Request) {
 // @Summary Get the api status
 // @Description Get the api status
 // @ID tide-whisperer-api-getstatus
-// @Accept json
 // @Produce json
-// @Success 200
-// @Failure 500 {string} string "error description"
+// @Success 200 {object} status.ApiStatus
+// @Failure 500 {object} status.ApiStatus
```
Comment (CedricPerotto) [inline data/api.go:107]: The Status struct is not renderered in the final swagger definition
```yaml
 status.ApiStatus:
    properties:
      status:
        type: Status
      version:
        type: string
    type: object
```

Comment (BigFab) [reply data/api.go:107]: indeed it is not. I did not find a way to make it. I think this is a drawback of the tool with deep models.

Comment (BigFab) [reply data/api.go:107]: using _--parseDependency_ in the _init_ command is making the trick

**Pull request merged**